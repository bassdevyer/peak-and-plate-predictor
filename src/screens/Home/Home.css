.Home #platePrefix {
    text-transform: uppercase;
}