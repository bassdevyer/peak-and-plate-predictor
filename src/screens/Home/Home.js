import React, { Component } from 'react'
import './Home.css'
import {
  ControlLabel,
  FormGroup,
  InputGroup,
  Panel,
  Image,
} from 'react-bootstrap'
import PlatePrefixFormControl
  from '../../components/PlatePrefixFormControl'
import PlateSuffixFormControl from '../../components/PlateSuffixFormControl'
import DateTimeFormControl from '../../components/DateTimeFormControl'
import ResultLabel from '../../components/ResultLabel'

export default class Home extends Component {

  constructor (props, context) {
    super(props, context)

    this.state = {
      platePrefix: '',
      plateSuffix: '',
      dateTime: '',
    }
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value,
    })
  }

  checkIfCanBeOnRoad = () => {
    let result = {}
    const negativeResult = { style: 'danger', label: 'No puede circular' }
    const positiveResult = { style: 'success', label: 'Puede circular' }
    const defaultResult = { style: 'default', label: 'Complete los datos' }
    if (this.state.platePrefix.length === 3
      && this.state.plateSuffix.length >= 3
      && this.state.dateTime) {
      const dateTime = new Date(this.state.dateTime)
      const dayOfWeekIndex = dateTime.getDay()
      const morningStartTime = new Date(this.state.dateTime).setHours(7, 0, 0)
      const morningEndTime = new Date(this.state.dateTime).setHours(9, 30, 0)
      const afternoonStartTime = new Date(this.state.dateTime).setHours(16, 0,
        0)
      const afternoonEndTime = new Date(this.state.dateTime).setHours(19, 30, 0)

      // Check if it's a working day
      if (dayOfWeekIndex > 0
        && dayOfWeekIndex < 6
        // If so, check if time is between 07:00 to 09:30 or 16:00 to 19:30
        && ((dateTime >= morningStartTime && dateTime <= morningEndTime)
          ||
          (dateTime >= afternoonStartTime && dateTime <= afternoonEndTime))) {
        const plateLastDigit = this.state.plateSuffix.slice(-1)
        switch (dayOfWeekIndex) {
          // Monday
          case 1:
            result = (plateLastDigit === '1' || plateLastDigit === '2')
              ? negativeResult
              : positiveResult
            break
          // Tuesday
          case 2:
            result = (plateLastDigit === '3' || plateLastDigit === '4')
              ? negativeResult
              : positiveResult
            break
          // Wednesday
          case 3:
            result = (plateLastDigit === '5' || plateLastDigit === '6')
              ? negativeResult
              : positiveResult
            break
          // Thursday
          case 4:
            result = (plateLastDigit === '7' || plateLastDigit === '8')
              ? negativeResult
              : positiveResult
            break
          // Friday
          case 5:
            result = (plateLastDigit === '9' || plateLastDigit === '0')
              ? negativeResult
              : positiveResult
            break
          default:
            result = positiveResult
        }
      } else {
        result = positiveResult
      }
    } else {
      result = defaultResult
    }
    return result
  }

  render () {
    const result = this.checkIfCanBeOnRoad()
    return (
      <Panel bsStyle="primary">
        <Panel.Heading>
          <Panel.Title componentClass="h3">
            Peak and Plate Predictor
          </Panel.Title>
        </Panel.Heading>
        <Panel.Body>
          <form className={'Home'}>
            <Image src="/info.png" responsive/>
            <FormGroup>
              <ControlLabel>Registration Plate</ControlLabel>
              <InputGroup>
                <PlatePrefixFormControl id='platePrefix'
                                        value={this.state.platePrefix}
                                        onChange={this.handleChange}/>
                <InputGroup.Addon>-</InputGroup.Addon>
                <PlateSuffixFormControl id='plateSuffix'
                                        value={this.state.plateSuffix}
                                        onChange={this.handleChange}/>
              </InputGroup>
            </FormGroup>
            <FormGroup>
              <ControlLabel>Date & Time</ControlLabel>
              <InputGroup>
                <DateTimeFormControl
                  id='dateTime'
                  value={this.state.dateTime}
                  onChange={this.handleChange}/>
              </InputGroup>
            </FormGroup>
            <FormGroup>
              <ResultLabel {...result}/>
            </FormGroup>

          </form>
        </Panel.Body>
      </Panel>
    )
  }
}