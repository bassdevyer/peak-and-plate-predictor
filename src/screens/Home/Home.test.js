// Link.react.test.js
import React from 'react'
import Home from '../Home'
import { create, } from 'react-test-renderer'
import ReactDOM from 'react-dom'

describe('Component Mounting', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Home/>, div)
  })
})

describe('Component default rendering', () => {
  test('Renders default component', () => {
    const component = create(
      <Home/>,
    )
    let instance = component.getInstance()
    expect(instance.state.platePrefix).toBe('')
    expect(instance.state.plateSuffix).toBe('')
    expect(instance.state.dateTime).toBe('')

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
