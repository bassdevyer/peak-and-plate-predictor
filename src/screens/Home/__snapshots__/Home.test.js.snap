// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default rendering Renders default component 1`] = `
<div
  className="panel panel-primary"
>
  <div
    className="panel-heading"
  >
    <h3
      className="panel-title"
    >
      Peak and Plate Predictor
    </h3>
  </div>
  <div
    className="panel-body"
  >
    <form
      className="Home"
    >
      <img
        className="img-responsive"
        src="/info.png"
      />
      <div
        className="form-group"
      >
        <label
          className="control-label"
        >
          Registration Plate
        </label>
        <span
          className="input-group"
        >
          <input
            className="platePrefixFormControl form-control"
            id="platePrefix"
            maxLength={3}
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value=""
          />
          <span
            className="input-group-addon"
          >
            -
          </span>
          <input
            className="plateSuffixFormControl form-control"
            id="plateSuffix"
            maxLength={4}
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value=""
          />
        </span>
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label"
        >
          Date & Time
        </label>
        <span
          className="input-group"
        >
          <input
            className="form-control"
            id="dateTime"
            onChange={[Function]}
            type="datetime-local"
            value=""
          />
        </span>
      </div>
      <div
        className="form-group"
      >
        <span
          className="label label-default"
        >
          Complete los datos
        </span>
      </div>
    </form>
  </div>
</div>
`;
